.github/workflows/bionic_build.yml

```diff
@@ -0,0 +1,51 @@
+name: Bionic-Build
+
+on:
+  push:
+  pull_request:
+  schedule:
+    - cron: '0 5 * * *'
+
+jobs:
+  industrial_ci:
+    name: Melodic
+    runs-on: ubuntu-latest
+    env:
+      CI_NAME: Bionic-Build
+      OS_NAME: ubuntu
+      OS_CODE_NAME: bionic
+      ROS_DISTRO: melodic
+      ROS_REPO: main
+      TARGET_WORKSPACE: '. github:ros-industrial/cmake_common_scripts#master github:Jmeyer1292/opw_kinematics#master'
+      ROSDEP_SKIP_KEYS: "iwyu cmake_common_scripts"
+      CCACHE_DIR: "/home/runner/work/descartes_light/descartes_light/Bionic-Build/.ccache"
+      TARGET_CMAKE_ARGS: "-DCMAKE_BUILD_TYPE=Debug -DDESCARTES_ENABLE_TESTING=ON"
+      AFTER_SCRIPT: 'catkin build -w $target_ws --no-deps --verbose descartes_ikfast --make-args test'
+    steps:
+      - uses: actions/checkout@v1
+
+      - name: Free Disk Space
+        run: |
+          sudo swapoff -a
+          sudo rm -f /swapfile
+          sudo apt clean
+          docker rmi $(docker image ls -aq)
+          df -h
+
+      - name: Prepare ccache timestamp
+        id: ccache_cache_timestamp
+        shell: cmake -P {0}
+        run: |
+          string(TIMESTAMP current_date "%Y-%m-%d-%H;%M;%S" UTC)
+          message("::set-output name=timestamp::${current_date}")
+
+      - name: ccache cache files
+        uses: actions/cache@v1.1.0
+        with:
+          path: ${{ env.CI_NAME }}/.ccache
+          key: ${{ env.CI_NAME }}-ccache-${{ steps.ccache_cache_timestamp.outputs.timestamp }}
+          restore-keys: |
+            ${{ env.CI_NAME }}-ccache-
```

Comment:
Out of curiosity, what is the point of freeing disk space and using `ccache` for a small project like this? My understanding of `ccache` was that it helps increase the speed of builds only after the project has been built once and a subset of the source code has been changed. Doesn't CI only build the repository one time?

Reply:
The clearing of the disk has little affect here but the cache works a little different. The github actions will upload the cache as an artifact and when a new PR is created it will pull this cache and use it to speed up build time. The artifact is specific to the CI_NAME so it is unique for a given CI workflow.